### PR TITLE
Add support to enable rgw

### PR DESCRIPTION
**Pull request by hemanthnakkina**
Update microceph terraform plans to add
integrations to identity service and
traefik-route-rgw.
Redeploy microceph once openstack plan is
deployed. This is to pick up the ingress-traefik-rgw and keystone-endpoints remote offers from openstack plan and integrate microceph with the latest available offers.

The offers from openstack plan is passed as
variables to microceph plan instead of retrieving
them directly in terraform plan from openstack
backend as the terraform plan fails if the openstack backend does not exist and this cannot be handled
in try clause.

Depends On:
- [x] https://github.com/canonical/microceph/pull/363
- [x] https://github.com/canonical/charm-microceph/pull/86
- [x] https://github.com/canonical/charm-microceph/pull/87
- [x] https://github.com/canonical/sunbeam-terraform/pull/76
- [x] https://github.com/canonical/sunbeam-terraform/pull/78

### CODE REVIEW

File: sunbeam-python/sunbeam/provider/local/commands.py [hunk 1]
```diff
@@ -481,6 +481,17 @@ def bootstrap(
                 proxy_settings=proxy_settings,
             )
         )
+        plan4.append(
+            DeployMicrocephApplicationStep(
```
Comment (gboutry) [inline sunbeam-python/sunbeam/provider/local/commands.py:485]: Can we add a comment about why this is needed ?

This is not immediately obvious that you want to fill in the relation offers

File: sunbeam-python/sunbeam/provider/maas/commands.py [hunk 2]
```diff
@@ -610,6 +610,17 @@ def deploy(
             proxy_settings=proxy_settings,
         )
     )
+    plan2.append(
+        DeployMicrocephApplicationStep(
```
Comment (gboutry) [inline sunbeam-python/sunbeam/provider/maas/commands.py:614]: Same here

**Review by gboutry — CHANGES_REQUESTED** [review]
This is looking good, just requesting comments, because re-applying a step is not obvious, and I might get crazy in the future when I forget about this change :smile:

File: sunbeam-python/sunbeam/provider/local/commands.py (hunk 1 above)
Comment (hemanthnakkina) [reply sunbeam-python/sunbeam/provider/local/commands.py:488]: Done

File: sunbeam-python/sunbeam/provider/maas/commands.py (hunk 2 above)
Comment (hemanthnakkina) [reply sunbeam-python/sunbeam/provider/maas/commands.py:617]: Done

**Comment by hemanthnakkina** [comment]
> This is looking good, just requesting comments, because re-applying a step is not obvious, and I might get crazy in the future when I forget about this change 😄

Completely agree, I thought of adding comment and somehow faded away. Now it is added, lets wait for CI

**Review by gboutry — APPROVED** [review]

**Pull request merged**